osu.Desktop/DiscordRichPresence.cs

```diff
@@ -173,6 +189,37 @@ private string truncate(string str)
             return null;
         }
 
+        private void fetchBeatmapSet()
+        {
+            IBeatmapInfo? beatmap = getBeatmap(activity.Value);
+            if (beatmap == null)
+            {
+                beatmapSetOnline.Value = null;
+                return;
+            }
+
+            if (beatmap.BeatmapSet is IBeatmapSetOnlineInfo online)
+            {
+                beatmapSetOnline.Value = online;
+            }
+            else
+            {
+                if (api.IsLoggedIn)
+                {
+                    var req = new GetBeatmapSetRequest(beatmap.OnlineID, BeatmapSetLookupType.BeatmapId);
+                    req.Success += res =>
+                    {
+                        beatmapSetOnline.Value = res;
+                    };
```

Comment:
What happens if this request fails, or even takes a while to complete? To me it looks that both scenarios may cause stale backgrounds from previous maps to be displayed.

This of course is caused by having to juggle both `activity` and `beatmapSetOnline`, and another argument for not bothering with the API call.

Reply:
This is resolved using the Beatmap Cache.

Discord won't immediately update the icon anyways, so a small delay between starting and showing the image isn't a major issue. Using the hardcoded url would simplify this however and will make it faster.